cf/delete-sample-file-lambda.yaml

```diff
@@ -0,0 +1,141 @@
+
+Parameters:
+  Environment:
+    Type: String
+    Default: development
+    AllowedValues:
+      - development
+      - staging
+      - production
+    Description: The environment for which the lambda is created
+
+Outputs:
+  DeleteSampleFileLambdaArn:
+    Description: ARN of DeleteSampleFileLambdaArn
+    Value: !GetAtt DeleteSampleFileLambda.Arn
+    Export:
+      Name: !Sub DeleteSampleFileLambdaArn-${Environment}
+
+Resources:
+  DeleteSampleFileLambdaRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: "sts:AssumeRole"
+      Path: /
+      Policies:
+        - PolicyName: !Sub "can-delete-sample-files-in-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 's3:DeleteObject'
+                Resource:
+                  - Fn::Sub: "arn:aws:s3:::biomage-originals-${Environment}-${AWS::AccountId}/*"
+        - PolicyName: !Sub "can-create-log-group-in-cloudwatch-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "logs:CreateLogGroup"
+                Resource:
+                  - Fn::Sub: "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/delete-sample-file-lambda-${Environment}"
+        - PolicyName: !Sub "can-add-logs-in-cloudwatch-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "logs:CreateLogStream"
+                  - "logs:PutLogEvents"
+                Resource:
+                  - Fn::Sub: "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/delete-sample-file-lambda-${Environment}:*"
+
+  DeleteSampleFileLambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      FunctionName: !Sub delete-sample-file-lambda-${Environment}
+      Runtime: nodejs14.x
+      Role: !GetAtt DeleteSampleFileLambdaRole.Arn
+      Handler: index.handler
+      Environment:
+        Variables:
+          AWS_ACCOUNT_ID: !Sub "${AWS::AccountId}"
+          CLUSTER_ENV: !Ref Environment
+      Code:
+        ZipFile: |
+          const AWS = require('aws-sdk');
+
+          const deleteSampleFile = (s3_path, bucketName) => {
+            const params = {
+              Bucket: bucketName,
+              Key: s3_path,
+            };
+
+            const s3 = new AWS.S3({
+              apiVersion: '2006-03-01',
+            });
+
+            s3.deleteObject(params, (err) => {
+              if (err) {
+                console.log(err, err.stack);
+                return;
+              }
```

Comment:
If the deletion fails is there any relatively easy way we can make this reach us?

Reply:
Slack notif?

Reply:
Or email, I don't know what would be easier to do

Reply:
It turns out that adding Slack notification here is not trivial. This might be expanded to a better way to send error notification from Lambda to Slack.  I've created a separate ticket for this: https://biomage.atlassian.net/browse/BIOMAGE-2026